src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,84 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int INITIAL_CAPACITY = 16;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfNeeded();
+        int index = getIndexFromHash(key);
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null);
+        } else {
+            Node<K, V> currentNode = table[index];
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    break;
+                }
+                currentNode = currentNode.next;
+            }
+            currentNode.next = new Node<>(key, value, null);
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = table[getIndexFromHash(key)];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resizeIfNeeded() {
+        if (size >= table.length * LOAD_FACTOR) {
+            size = 0;
+            Node<K, V>[] oldTable = table;
+            table = (Node<K, V>[]) new Node[table.length * 2];
+            for (Node<K, V> node : oldTable) {
+                while (node != null) {
+                    put(node.key, node.value);
+                    node = node.next;
+                }
+            }
+        }
+    }
+
+    private int getIndexFromHash(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
```

Comment:
You don't need additional brackets 
```suggestion
        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
```